src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -135,6 +183,47 @@ static int GetUnescapedOffset(ReadOnlySpan<char> fullName, int startIndex)
             }
         }
 
+        internal static string Unescape(string input)
+        {
+            int indexOfEscapeCharacter = input.IndexOf(EscapeCharacter);
+            if (indexOfEscapeCharacter < 0)
+            {
+                // Nothing to escape, just return the original value.
+                return input;
+            }
+
+            return UnescapeToBuilder(input, indexOfEscapeCharacter);
+
+            static string UnescapeToBuilder(string name, int indexOfEscapeCharacter)
+            {
+                // this code path is executed very rarely (IL Emit or pure IL with chars not allowed in C# or F#)
+                var sb = new ValueStringBuilder(stackalloc char[64]);
+                sb.EnsureCapacity(name.Length);
+                sb.Append(name.AsSpan(0, indexOfEscapeCharacter));
+
+                for (int i = indexOfEscapeCharacter; i < name.Length;)
+                {
+                    char c = name[i++];
+
+                    if (c != EscapeCharacter || i == name.Length)
+                    {
+                        sb.Append(c);
+                    }
+                    else if (name[i] == EscapeCharacter) // escaped escape character ;)
+                    {
+                        sb.Append(c);
+                        // Consume the escaped escape character, it's important for edge cases
+                        // like escaped escape character followed by another escaped char (example: "\\\\\\+")
+                        i++;
+                    }
+                }
+
+                string result = sb.ToString();
+                sb.Dispose();
+                return result;
```

Comment:
```suggestion
                return sb.ToString();
```
ValueStringBuilder.ToString calls Dispose

Reply:
Done.